python/tank/platform/software_launcher.py

```diff
@@ -0,0 +1,406 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Defines the base class for DCC application launchers all Tank engines
+should implement.
+"""
+
+import os
+import sys
+import re
+import logging
+import xml.etree.ElementTree as XML_ET
+
+from ..errors import TankError
+from ..log import LogManager
+from ..util.loader import load_plugin
+
+from . import constants
+from . import validation
+
+from .bundle import resolve_setting_value
+from .engine import get_env_and_descriptor_for_engine
+
+# std core level logger
+core_logger = LogManager.get_logger(__name__)
+
+def create_engine_launcher(tk, context, engine_name):
+    """
+    Factory method that creates a Toolkit engine specific
+    SoftwareLauncher instance.
+
+    :param tk: :class:`~sgtk.Sgtk` Toolkit instance.
+    :param context: :class:`~sgtk.Context` Context to launch the DCC in.
+    :param engine_name: (str) Name of the Toolkit engine associated with
```

Comment:
For builtin data types you can do:

`:param str engine_name:`

Sphinx will do the right thing.